src/vcpkg/build.cpp

```diff
@@ -1320,15 +1327,16 @@ namespace vcpkg
         auto& abi_info = action.abi_info.value_or_exit(VCPKG_LINE_INFO);
```

Comment:
As a drive-by improvement, it would be nice to deduplicate this logic a bit:

```cpp
ExtendedBuildResult result =
    do_build_package_and_clean_buildtrees(args, paths, action, all_dependencies_satisfied);

auto& abi_info = action.abi_info.value_or_exit(VCPKG_LINE_INFO);
if (auto p_abi_file = abi_info.abi_tag_file.get()) {
    // do the binary caching things
}
return result;
```